src/runtime_src/core/edge/user/aie/aie.cpp

```diff
@@ -417,4 +418,117 @@ start_profiling_event_count(const std::string& port_name)
   return handle;
 }
 
+void
+Aie::
+configure_bd(int tileType, uint8_t column, uint8_t row, uint8_t bdId,
+  uint64_t address,
+  uint32_t length,
+  const std::vector<uint32_t>& stepsize,
+  const std::vector<uint32_t>& wrap,
+  const std::vector<std::pair<uint32_t, uint32_t>>& padding,
+  bool enable_packet,
+  uint8_t packet_id,
+  uint8_t out_of_order_bd_id,
+  bool tlast_suppress,
+  uint32_t iteration_stepsize,
+  uint16_t iteration_wrap,
+  uint8_t iteration_current,
+  bool enable_compression,
+  bool lock_acq_enable,
+  int8_t lock_acq_value,
+  uint8_t lock_acq_id,
+  int8_t lock_rel_value,
+  uint8_t lock_rel_id,
+  bool use_next_bd,
+  uint8_t next_bd,
+  uint8_t burst_length)
+{
+  if (!devInst)
+    throw xrt_core::error(-EINVAL, "Can't configure bd: AIE is not initialized");
+
+  if (access_mode == xrt::aie::access_mode::shared)
+    throw xrt_core::error(-EPERM, "Shared AIE context can't configure bd");
+
+  adf::dma_api::configureBD(tileType, column, row, bdId,
```

Comment:
We have to let adf know we will not support this kind of API.  This the worst function signature I have seen in a very long time. @jerryhsu-xilinx